apps/client/src/metrics/cloud-watch-metrics-recorder.ts

```diff
@@ -0,0 +1,125 @@
+import { type MetricsRecorder } from './metrics-recorder.interface';
+import {
+  CloudWatchClient,
+  type MetricDatum,
+  PutMetricDataCommand,
+  type PutMetricDataCommandInput,
+} from '@aws-sdk/client-cloudwatch';
+import throttle from 'lodash/throttle';
+import { type AuthService } from '~/auth/auth-service.interface';
+import { authService } from '~/auth/auth-service';
+
+// at most one putMetrics() call every X ms
+const PUT_METRICS_THROTTLE = 3000;
+
+type Metric = Parameters<MetricsRecorder['record']>[0];
+
+class CloudWatchMetricsQueue {
+  private readonly queue = new Set<MetricDatum>();
+
+  enqueue(metric: Metric) {
+    const datum = CloudWatchMetricsQueue.parseMetric(metric);
+    this.queue.add(datum);
+  }
+
+  getMetricData() {
+    return Array.from(this.queue);
+  }
+
+  clear() {
+    this.queue.clear();
+  }
+
+  private static parseMetric({
```

Comment:
Why is this method static, if it is private and only referenced within the class by non-static public method?

Reply:
yes, it is referenced within the class only, mostly for code structure reason. They don't have to live under `cloud-watch-metrics-recorder.ts` but there's no need to use those methods elsewhere currently.